### ISSUE TITLE
`Generic.PHP.Syntax` false negative when file content passed via STDIN

### ISSUE THREAD

**Issue by rodrigoprimo** [PHPCSStandards/PHP_CodeSniffer#915]
## Describe the bug

The `Generic.PHP.Syntax` sniff is unable to check for syntax errors when the file content is passed via STDIN, resulting in a false negative.

### To reproduce
1. Run `echo '<?php $array = [1, 2, 3; // Missing closing bracket.' | phpcs --standard=Generic --sniffs=Generic.PHP.Syntax -`
3. No errors are displayed.

## Expected behavior

I believe the command above should display an error as there is a syntax error in the provided PHP code (missing closing bracket).

## Additional information

This is happening because the sniff does not consider that `$phpcsFile->getFilename()` might return `STDIN` instead of a file name:

https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/dd80edb4e91e58b2bb423d361ec8281aef44cfa6/src/Standards/Generic/Sniffs/PHP/SyntaxSniff.php#L59-L60

When `STDIN` is passed to the `php -l` command, PHP returns an error saying that the file does not exist, and the sniff ends its execution without generating an error.

I think the sniff could be modified to run a different command when the file content is passed via STDIN (I'm not sure if this would work on Windows):

```
echo '<?php $array = [1, 2, 3; // Missing closing bracket.' | /usr/bin/php7.4 -l -d display_errors=1 -d error_prepend_string='' 2>&1
```

## Versions (please complete the following information)

|                                       |                                                                                                           |
|--------------------------|----------------------------------------------------------------------  |
| Operating System           | Linux                                                       |
| PHP version                    | 8.3                                                                                     |
| PHP_CodeSniffer version | master                                                                            |
| Standard                         | Generic                                                      |
| Install type                      | git clone |

## Please confirm

- [x] I have searched the issue list and am not opening a duplicate issue.
- [x] I have read the [Contribution Guidelines](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/.github/CONTRIBUTING.md) and this is not a [support question](https://github.com/PHPCSStandards/PHP_CodeSniffer/discussions).
- [x] I confirm that this bug is a bug in PHP_CodeSniffer and not in one of the external standards.
- [x] I have verified the issue still exists in the `master` branch of PHP_CodeSniffer.

**Comment by jrfnl** [comment]
@rodrigoprimo I believe there is still something wrong in the command as the tests do not pass on Windows.

Aside from that, could you please rebase this branch on the current `master` ?
